src/material.js

```diff
@@ -172,6 +172,19 @@ export class Material {
         return new Lattice(this.lattice);
     }
 
+    getDerivedPropertiesForHash() {
```

Comment:
Moving the derivedProperty functions from the materials class in the web-app to made-js caused errors in hash generation due to initial materials being generated containing no derived properties.